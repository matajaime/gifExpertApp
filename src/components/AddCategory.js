import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({addCategory}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        //console.log(event);
        if ( inputValue != '' ) {
            addCategory(inputValue);
            setInputValue('');    
        }
    };

    const cleanInput = (event) => {
        event.target.value = '';
    }

    return (
        <form onSubmit={handleSubmit}>
        <h4>Add Category</h4>
        <input type="text" value={inputValue} placeholder="Ingresa una categoria" 
        onChange={handleInputChange} onFocus={cleanInput}/>
        </form>
    );
};

AddCategory.propTypes = {
    addCategory: PropTypes.func.isRequired
}
