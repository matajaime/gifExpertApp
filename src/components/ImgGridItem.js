import React from "react";
import PropTypes from 'prop-types';

export const ImgGridItem = ({img}) => {
    //console.log(img);
    return (
        <div className="card animate__animated animate__fadeInDownBig">
            <img src={img.url} alt={img.title} />
            <p>{img.title}</p>
        </div>
    );
};

ImgGridItem.propTypes = {
    img : PropTypes.object.isRequired
}

