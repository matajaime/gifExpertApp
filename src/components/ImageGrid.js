import React, {useEffect, useState} from "react";
import { ImgGridItem } from "./ImgGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const ImageGrid = ({category}) => {
    
    //const[gifs, setGifs] = useState([]);
    //useEffect( () => {
    //    getGifs(category).then( setGifs );
    //}, [ category ]);

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
        <h3>{category}</h3>
        { loading ? <p>Cargando...</p> : 
        <div className="card-grid">
            { images.map((gif)=>{
                return <ImgGridItem key={gif.id} img={gif} />;
            })}
        </div>}
        </>
    )
};

ImageGrid.propTypes = {
    category: PropTypes.string.isRequired
}
