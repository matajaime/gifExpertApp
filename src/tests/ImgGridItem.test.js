import React from 'react';
import { shallow } from 'enzyme';
import { ImgGridItem } from '../components/ImgGridItem';

describe('ImgGridItem tests', ()=>{
    const img = {
        url: 'MiLiga',
        title: 'MiTitulo'
    };
    const wrapper = shallow(<ImgGridItem img={img} />);
    test('SnapShot of init state of GridItem',() => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Title should match the paragraph',()=>{
        const paragraph = wrapper.find('p').text().trim();
        expect(paragraph).toBe(img.title);
    });
    test('Image source should match url', () => {
        const imgSrc = wrapper.find('img').props().src;
        expect(imgSrc).toBe(img.url);
    });
    test('Component should have animate__fadeInDownBig effect',()=>{
        const div = wrapper.find('div');
        const actual = div.props().className.indexOf('animate__fadeInDownBig') >= 0;
        expect( actual ).toBe( true );
    })
});