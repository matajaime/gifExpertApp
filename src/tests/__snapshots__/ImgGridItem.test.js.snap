// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgGridItem tests SnapShot of init state of GridItem 1`] = `
<div
  className="card animate__animated animate__fadeInDownBig"
>
  <img
    alt="MiTitulo"
    src="MiLiga"
  />
  <p>
    MiTitulo
  </p>
</div>
`;
